discovery_engine_core/web-api/src/models.rs

```diff
@@ -79,27 +83,30 @@ impl From<Document> for Article {
 /// Represents user interaction request body.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub(crate) struct InteractionRequestBody {
-    pub(crate) document_id: Id,
+    pub(crate) document_id: String,
 }
 
 /// Unique identifier for the user.
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Display)]
 pub(crate) struct UserId(String);
 
 impl UserId {
-    fn new(user_id: &str) -> Result<Self, Error> {
-        if user_id.is_empty() {
-            Err(Error::EmptyUserId)
+    fn new(value: &str) -> Result<Self, Error> {
+        let value = value.trim();
```

Comment:
I just tested it and it looks like that the string that wrap returns is url encoded :(
that means `trim` won't help here 

e.g.
```
curl --request POST \
  --url http://0.0.0.0:3000/user/%20/interaction \
  --header 'Content-Type: application/json' \
  --data '{
  "document_id": "0251ae9f73ec12f4d3eced9c4dc9ccc8"
}'
```
is a valid request with the user_id `%20`

Reply:
ah, good catch! I've added this to the validations, could you check again?

Reply:
That works but now something like `0.0.0.0:3000/user/E%20W` is not valid anymore.

what about decoding first with `urlencoding::decode` and after that check if it the value is empty or contains a zero byte  

```rust 
        let value = urlencoding::decode(value).map_err(Error::Utf8Error)?;

        let value = value.trim();
        if value.is_empty() {
            Err(Error::UserIdEmpty)
        } else if value.contains('\u{0000}') {
            Err(Error::UserIdContainsZeroByte)
        } else {
            Ok(Self(value.to_string()))
        }
```

Reply:
yup, you're right :) I've changed it per your recommendation